FaunaDB.Client/Types/Decoder.cs

```diff
@@ -93,6 +95,72 @@ static object DecodeIntern(Value value, Type dstType)
                 throw invalidCast(value, dstType);
             }
 
+            if (forceType == typeof(StringV) && value.GetType() == typeof(StringV))
+            {
+                if (value != null && value != NullV.Instance)
+                    switch (Type.GetTypeCode(dstType))
+                    {
+                        case TypeCode.Object:
+                            if (typeof(DateTimeOffset) == dstType)
+                                return new DateTimeOffset(Convert.ToDateTime(ToString(value)));
+
+                            else if (typeof(byte[]) == dstType && value is BytesV)
+                                return JsonConvert.DeserializeObject<byte[]>(ToString(value));
+
+                            else if (dstType.IsArray && dstType.HasElementType)
+                                return JsonConvert.DeserializeObject(ToString(value), dstType);
+
+                            else if (typeof(IDictionary).IsAssignableFrom(dstType) || (dstType.GetTypeInfo().IsGenericType && dstType.Is(typeof(IDictionary<,>))))
```

Comment:
This switch/case is a repetition, also I have the impression you can inline the deserialization in the block that already exists below it, like:
```csharp
                    if (typeof(IDictionary).IsAssignableFrom(dstType) || (dstType.GetTypeInfo().IsGenericType && dstType.Is(typeof(IDictionary<,>))))
                    {
                        if (forceType == typeof(StringV) && value.GetType() == typeof(StringV))
                            return JsonConvert.DeserializeObject(ToString(value), dstType);

                        return FromDictionary(value, dstType);
                    }
```
and, given you are adding that "serialization" feature, I would like to see tests covering it (encoding and decoding).

Another thing, what would be the use case for this serialization/deserialization? I liked the part where you call a string constructor for objects like `Url` and `Guid`, but I'm not sure I like this json serialization/deserialization.

Reply:
How else would one represent collection like data as a string? Fauna String should work on most types if people want to store it as a string. If never store a collection like that, but somebody else might have a reason to put fauna string on a collection.

Reply:
How else would one represent collection like data as a string? Fauna String should work on most types if people want to store it as a string. If never store a collection like that, but somebody else might have a reason to put fauna string on a collection.

Reply:
Also I can take out the switch and put it all in the other block but there will be a lot of redundancies there too. You can't get around that here really. Like in every case there will be another if statement: if (forceType == typeof(StringV) && value.GetType() == typeof(StringV))

Reply:
actually not every case, just for `TypeCode.Object`, case like `TypeCode.Single`, `TypeCode.Double` and etc are handled automatically because `ToNumber` function return strings as well and the `Convert` functions accept strings

Reply:
I deleted the switch and moved it inline